students/Michael Hrstka/FINAL/pedal/pedal.ino

```diff
@@ -0,0 +1,91 @@
+int guitarInput = 0;
+int outputPins[5] = {9, 10, 11, 12, 13};
+bool distortion = 0;
+bool reverb = 0;
+
+int distortionBtn = 2;
+int reverbBtn = 3;
+
+int distortionLight = 4;
+int reverbLight = 5;
+
+int previousWave = 0;
+int reverbArray[300];
+int reverbIndex = 0;
+
+void setup() {
+  pinMode(outputPins[0], OUTPUT);
+  pinMode(outputPins[1], OUTPUT);
+  pinMode(outputPins[2], OUTPUT);
+  pinMode(outputPins[3], OUTPUT);
+  pinMode(outputPins[4], OUTPUT);
+
+  pinMode(distortionBtn, INPUT);
+  digitalWrite(distortionBtn, HIGH);
+
+  pinMode(reverbBtn, INPUT);
+  digitalWrite(reverbBtn, HIGH);
+
+  attachInterrupt(digitalPinToInterrupt(distortionBtn), toggleDistortion, FALLING);
+  attachInterrupt(digitalPinToInterrupt(reverbBtn), toggleReverb, FALLING);
+
+  pinMode(distortionLight, OUTPUT);
+  pinMode(reverbLight, OUTPUT);
+
+}
+
+void loop() {
+  double guitarVal = analogRead(guitarInput);
+
+  if (distortion) {
+    guitarVal *= 5;
+  }
+
+  double wave = guitarVal * 2 / 1023.0;
+
+  int newWave = wave * 16;
+  if (newWave > 31) {
+    newWave = 31;
+  }
+
+  /* Misguided initial attempt at adding reverb - Fixes crackling */
+  newWave = (newWave + previousWave) / 2; // Average wave
+  previousWave = newWave;
+
+  if (reverb) {
+    if (distortion){
+      reverbArray[reverbIndex == 0 ? 299 : reverbIndex - 1] = newWave / 4 + reverbArray[reverbIndex] / 4;
+      newWave = (newWave +  reverbArray[reverbIndex]) / 2;
+    } else {
+      newWave = (newWave +  reverbArray[reverbIndex]) / 2;
+      reverbArray[reverbIndex == 0 ? 299 : reverbIndex - 1] = newWave + reverbArray[reverbIndex] / 4;
+    }
+    reverbIndex = reverbIndex >= 299 ? 0 : reverbIndex + 1;
+  }
+
+  setOutput(newWave);
+}
+
+void toggleDistortion() {
+  distortion = !distortion;
+  digitalWrite(distortionLight, distortion);
+}
+
+void toggleReverb() {
+  reverb = !reverb;
+  digitalWrite(reverbLight, reverb);
+}
+
+void setOutput(int output) {
+    // Converting to binary between the pins
```

Comment:
With a little more design, you can just write your output int directly to an output port. Would save time.